tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit--help')).toHaveLength(2);
+  });
+
+  it('renders a displayName input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-displayName')).toHaveLength(1);
+  });
+
+  it('renders a disabled displayName input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-displayName'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a homepage input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveLength(1);
+  });
+
+  it('renders a disabled homepage input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveProp('disabled', true);
+  });
+
+  it('renders a location input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveLength(1);
+  });
+
+  it('renders a disabled location input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveProp('disabled', true);
+  });
+
+  it('renders a occupation input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveLength(1);
+  });
+
+  it('renders a disabled occupation input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveProp('disabled', true);
+  });
+
+  it('renders a biography input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-biography')).toHaveLength(1);
+  });
+
+  it('renders a disabled biography input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-biography')).toHaveProp('disabled', true);
+  });
+
+  it('dispatches editUserAccount action with all fields', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = mountUserProfileEdit({ errorHandler, store });
+    const user = getCurrentUser(store.getState().users);
+
+    root.find('.UserProfileEdit-form').simulate('submit', createFakeEvent());
+
+    sinon.assert.calledWith(dispatchSpy, editUserAccount({
```

Comment:
Is this one test with the default values good enough, or should we have (at least) another which sets actual values that are not the defaults and verifies that dispatch is called with the correct values?

Reply:
As a follow-up to this, are we happy to just be testing with an updated `location`, or should we test with all of the editable fields, either by setting all of them and submitting, or each one and then submitting individually in a loop?